packages/plugins/src/plugins/maker/helper.ts

```diff
@@ -0,0 +1,30 @@
+import { Address, PublicClient } from 'viem';
+import { EvmNetworkIdType } from '@sonarwatch/portfolio-core';
+import { proxyRegAbi } from './abis';
+import { proxyRegAddress } from './constants';
+import { getEvmClient } from '../../utils/clients';
+
+/**
+ * Fetches the proxy address for a given user address.
+ * @param userAddress - The address of the user.
+ * @returns The proxy address associated with the user.
+ */
+export const getProxyAddress = async (
+  userAddress: Address,
+  networkId: EvmNetworkIdType
+): Promise<Address | undefined> => {
+  const client: PublicClient = getEvmClient(networkId);
+  try {
+    const proxyAddress = await client.readContract({
+      address: proxyRegAddress as Address,
```

Comment:
Did you consider explicitly typing https://github.com/Octav-Labs/sonarwatch-portfolio/blob/7499f1c8a346a0260051b2128c1bd82ed30e0f0f/packages/plugins/src/plugins/maker/constants.ts#L10 as an `Address` to avoid the current forced-cast ?

```suggestion
      address: proxyRegAddress,
```

Reply:
i didnt want to touch charlies stuff too much but yeah i can